app/controllers/mixins/ems_common_angular.rb

```diff
@@ -432,15 +458,29 @@ def set_ems_record_vars(ems, mode = nil)
       build_connection(ems, endpoints, mode)
     end
 
-    def get_hostname_from_routes(ems, hostname, port, token)
+    def get_hostname_from_routes(ems, endpoint_hash, token)
       return nil unless ems.class.respond_to?(:openshift_connect)
-      client = ems.class.openshift_connect(hostname, port, :bearer => token)
+      endpoint = Endpoint.new(endpoint_hash)
+      ssl_options = {
+        :verify_ssl => ems.verify_ssl_mode(endpoint),
+        :cert_store => ems.ssl_cert_store(endpoint)
+      }
+      client = ems.class.raw_connect(endpoint.hostname, endpoint.port,
+                                     :service => :openshift, :bearer => token, :ssl_options => ssl_options)
       client.get_route('hawkular-metrics', 'openshift-infra').try(:spec).try(:host)
     rescue KubeException => e
       $log.warn("MIQ(#{controller_name}_controller-#{action_name}): get_hostname_from_routes error: #{e}")
       nil
     end
 
+    def container_security_options(security_protocol, certificate_authority)
+      {
+        :security_protocol     => security_protocol,
+        :verify_ssl            => %w(ssl-without-validation non-ssl).exclude?(security_protocol),
```

Comment:
Nice.  Not all providers will use same logic, but good to share.
Let's rename the function to something more generic, say `endpoint_security_options`?